### DISCUSSION TITLE
Muliple groops version on one PC

### DISCUSSION THREAD

**Discussion by Bemtgenj** [groops-devs/groops#186] [Q&A]
Hello,

If we want to run different version of groops on one PC (i.e baseline groops version, DTM included groops version, changed star camera script groops version,...) - Is it as simple as for example naming the different groops directories with a different name (i.e groopsbaseline, groop dtm, groopsstarcam,...) and then changing that specific groops directory name to groops (as that is the default directory name for groops to run) for the version that you want to use for the run. Maybe for Linux run .bashrc again?

We just wanted to make sure that the correct versions is ran at any given time . This is what we were  thinking should work, we but just we wanted to be sure that we did not miss anything

Txs for the help and feedback

Cheers
Jean

**Comment by Waley04** [comment]
@Bemtgenj,

**Windows:**
 In my case, I installed different directories on the same PC. I, then, simply specified the full path in the System Environment Variables. I also have a different version installed on Windows Subsystem for Linux on the same PC. This is more convenient as I can use a common directory for GNSS datasets when undertaking batch processing. 

**Linux:**
 I use version alias. I renamed the GROOPS software by tagging the version number. For example, I have 'groopsOne', 'groopsTwo', ....

===> Or, you may modify some libraries or configuration files before compiling them in order for the software to appear with a version of choice.

Cheers.

**Comment by Bemtgenj** [comment]
Hi waley

Thank you very much for you feedback,  i will set it up with different aliases instead of renaming the directories every time when i run it. This makes it easier as you run the proper version and do not have to remember to rename the directory.

Cheers
Jean

**Comment by Bemtgenj** [comment]
Hello

we were thinking of another method but somehow i did not get it to work. groops has a simulateCamera script inherent in the software. We modified this simulateCamera script to simulateCameraCubesat script and added in the same directory as the original one and then the idea was to recompile groops with then having the option of both being available when setting up the scripts. When i recompiled groops it ignore this simulateCameraCubesat (i checked in the processing of the compilation and all the other *.cpp did show up). I am not sure what i did wrong. Is there a list somewhere that groops uses to use the correct files for the compilation?

Txs for the help

Cheers
Jean

**Reply by tmayerguerr** [reply]
You have to add the cpp file in source/sourcesCXX.txt
regards
Torsten

**Comment by Bemtgenj** [comment]
Hi Torsten

txs very much for the quick feedback

Cheers
jean

**Comment by Bemtgenj** [comment]
Hi Torsten,

I have finally gotten around to doing this change, but ran into a small issue that once i did it the GUI did not show this new additional script - Is there another list that i need to update or is it enough that the source/sourcesCXX.txt is changed and the problem came from somewhere else?

I found a program list variable in the cpp files but i am not sure if this is the program listing and where it is populated?
Txs for the help and the feedback
Cheers
Jean

**Reply by tmayerguerr** [reply]
Hi Jean,

The GUI reads a XML schema file with all the options. The file is created by the command:
       ```groops.exe --xsd C:\groops\groops.xsd```
To update the html documentation, run:
       ```groops.exe --doc C:\groops\docs\```

regards
Torsten

**Comment by Bemtgenj** [comment]
Hi Torsten

Thank you very much for the quick feedback. i will run those 2 commands

Cheers
Jean
